api/proto/config/storage.proto

```diff
@@ -242,3 +243,48 @@ message DisksConfig {
   DisksArrayType array_type = 2;
   repeated DisksConfig children = 3;
 }
+
+// SnapshotOpsCmd counter for snapshot cmd
+message SnapshotOpsCmd {
+  uint32 counter = 1;
+  google.protobuf.Timestamp opsTime = 2;
+}
+
+// ZfsSnapshot describes a snapshot for a specific logical
+// volume that must exist on the device. It has a required
+// field volume_uuid for which it was created.
+message ZfsSnapshot {
+  // The real snapshot name in ZFS
+  string id = 1;
+
+  // Display name (User-friendly name). This name does not affect
+  // the snapshot properties in ZFS in any way
+  string display_name = 2;
+
+  // ID of the volume this snapshot belongs to. The field is required
+  // for all commands except for the rename command.
+  string volume_uuid = 3;
+
+  // Time of the last operation on the snapshot
+  uint64 last_cmd_timestamp = 4;
+
+  // Flag indicating that the snapshot has been deleted
+  bool deleted = 5;
+
+  // The command for creating a snapshot, for execution, requires the presence
+  // of a volume_uuid for which the snapshot will be created. By default, populates
+  // display_name with the date the snapshot was taken.
+  SnapshotOpsCmd create = 6;
+
+  // The command to delete a snapshot in ZFS.
+  SnapshotOpsCmd destroy = 7;
+
+  // The command to change the value for display_name.
+  SnapshotOpsCmd rename = 8;
+
+  // The command for the snapshot rollback operation.
+  // Keep in mind that in order to return to any of the
+  // previous snapshots, all intermediate snapshots will be
+  // destroyed in ZFS. These snapshots must be marked as deleted.
```

Comment:
I think in that case we should set `deleted` flag in info message, correct?

Reply:
Sounds logical, but why can't we just send only those snapshots that really exist in informational messages? (Because ZFS doesn't keep a list of deleted snapshots.)

I believe that information messages should reflect the current situation, in which there will be only up-to-date data.

Why do we need to store a history of deleted snapshots on the EVE side, if we can answer the request of the cloud that this snapshot has been deleted?

Can the cloud after the rollback operation automatically mark all the snapshots that were taken after the date of the snapshot to which we rolled back as deleted?

With the logic you suggested for the flag in the info message, the cloud can assume that all snapshots that didn't arrive in the message have been deleted, then the flag would make no sense.
It is discussible.